dipy/reconst/fwdti.py

```diff
@@ -0,0 +1,871 @@
+from dipy.core.ndindex import ndindex
+
+
+def fwdti_prediction(params, gtab, S0=1, Diso=3.0e-3):
+    """ Signal prediction given the free water DTI model parameters.
+
+    Parameters
+    ----------
+    params : ndarray
+        Model parameters. The last dimension should have the 12 tensor
+        parameters (3 eigenvalues, followed by the 3 corresponding
+        eigenvectors) and the volume fraction of the free water compartment.
+    gtab : a GradientTable class instance
+        The gradient table for this prediction
+    S0 : float or ndarray
+        The non diffusion-weighted signal in every voxel, or across all
+        voxels. Default: 1
+    Diso : float, optional
+        Value of the free water isotropic diffusion. Default is set to 3e-3
+        $mm^{2}.s^{-1}$. Please adjust this value if you are assuming different
+        units of diffusion.
+
+    Notes
+    -----
+    The predicted signal is given by:
+    $S(\theta, b) = S_0 * [(1-f) * e^{-b ADC} + f * e^{-b D_{iso}]$, where
+    $ADC = \theta Q \theta^T$, $\theta$ is a unit vector pointing at any
+    direction on the sphere for which a signal is to be predicted, $b$ is the b
+    value provided in the GradientTable input for that direction, $Q$ is the
+    quadratic form of the tensor determined by the input parameters, $f$ is the
+    free water diffusion compartment, $D_{iso}$ is the free water diffusivity
+    which is equal to $3 * 10^{-3} mm^{2}s^{-1} [1]_.
+
+    References
+    ----------
+    .. [1] Hoy, A.R., Koay, C.G., Kecskemeti, S.R., Alexander, A.L., 2014.
+           Optimization of a free water elimination two-compartmental model
+           for diffusion tensor imaging. NeuroImage 103, 323-333.
+           doi: 10.1016/j.neuroimage.2014.09.053
+    """
+    evals = params[..., :3]
+    evecs = params[..., 3:-1].reshape(params.shape[:-1] + (3, 3))
+    f = params[..., 12]
+    qform = vec_val_vect(evecs, evals)
+    lower_dt = lower_triangular(qform, S0)
+    lower_diso = lower_dt.copy()
+    lower_diso[..., 0] =  lower_diso[..., 2] = lower_diso[..., 5] = Diso
+    lower_diso[..., 1] =  lower_diso[..., 3] = lower_diso[..., 4] = 0
+    D = design_matrix(gtab)
+
+    pred_sig = np.zeros(f.shape + (gtab.bvals.shape[0],))
+    mask = _positive_evals(evals[..., 0], evals[..., 1], evals[..., 2])    
+    index = ndindex(f.shape)
+    for v in index:
+        if mask[v]:
+            pred_sig[v] = (1 - f[v]) * np.exp(np.dot(lower_dt[v], D.T)) + \
+                          f[v] * np.exp(np.dot(lower_diso[v], D.T))
+
+    return pred_sig
+
+
+class FreeWaterTensorModel(ReconstModel):
+    """ Class for the Free Water Elimination Diffusion Tensor Model """
+    def __init__(self, gtab, fit_method="NLS", *args, **kwargs):
+        """ Free Water Diffusion Tensor Model [1]_.
+
+        Parameters
+        ----------
+        gtab : GradientTable class instance
+        fit_method : str or callable
+            str can be one of the following:
```

Comment:
It's very similar to the docstring of the `TensorModel.__init__`, which *almost* renders fine. One more line-break here should do it, I believe.